### PR TITLE
Reduce usage of Apache Commons Lang 2

### LINKED ISSUE #3408: Reduce usage of Apache Commons Lang 2
Apache commons lang 2 is not maintained anymore since years. Usage of it should be replaced with Apache Commons Lang 3. Both libraries are part of AEM as a Cloud Service and AEM 6.x, only a change of the imports is required.

There is one exception, the usage of AEMs MailTemplate API which currently does not provide an alternative. Once that is available we can change the last remaining usage of Apache commons lang 2.

### PR DESCRIPTION
**Pull request by cziegeler**
Replace usage of Apache Commons Lang 2 with Apache Commons Lang 3
As noted in #3408 there are two places remaining (MailTemplate) which can be replaced as soon as there is an alternative.
This fixes #3408

### CODE REVIEW

**Comment by cziegeler** [comment]
@kwin You were right, we should convert the on-prem bundle into a fragment; I did this as part of this PR as we will most likely need this once an alternative for the MailTemplate is available.

**Comment by cziegeler** [comment]
@kwin Do you think we can already apply this? The remaining usage will be replaced once an alternative API is available for Cloud Service

File: bundle-onprem/pom.xml [hunk 1]
```diff
@@ -33,13 +33,18 @@
     <!-- ====================================================================== -->
 
     <artifactId>acs-aem-commons-bundle-onprem</artifactId>
-    <name>ACS AEM Commons Bundle - On Prem</name>
-    <description>Core ACS AEM Commons OSGi Bundle for AEM on prem only</description>
+    <name>ACS AEM Commons - Bundle Fragment (for On Prem</name>
```
Comment (kwin) [inline bundle-onprem/pom.xml:36]: This is missing a closing parenthesis

**Review by kwin — APPROVED** [review]
Looks good except for one small glitch in the Maven name.

File: bundle-onprem/pom.xml (hunk 1 above)
Comment (cziegeler) [reply bundle-onprem/pom.xml]: Thanks for spotting, fixed now

**Pull request merged**